README.md

```diff
@@ -72,7 +72,7 @@ func main() {
 	ctx := context.Background()
 
 	// Create a new client, with no features.
-	client := mcp.NewClient("mcp-client", "v1.0.0", nil)
+	client := mcp.NewClient(nil, nil)
```

Comment:
I don't think nil should be a valid first argument. We should probably panic in that case (and perhaps also if implementation.Name or Version are missing)? WDYT?